### PR TITLE
Update code.gitea.io/git

### PR DESCRIPTION
**Pull request by ethantkoenig**
Bring in latest changes from https://github.com/go-gitea/git. Includes fix for https://github.com/go-gitea/gitea/issues/491 and https://github.com/go-gitea/gitea/issues/502 (see https://github.com/go-gitea/git/pull/53)

### CODE REVIEW

**Comment by lunny** [comment]
@ethantkoenig since some function params changed, you have to change the invoke places.

**Comment by ethantkoenig** [comment]
I'm confused as to why `make test-vendor` is failing in the CI; it passes when I run it locally

**Comment by sapk** [comment]
@ethantkoenig you fetch the commit of your PR that is not the same as the one in master ~~+ you edit file after~~.
Please, just do `govendor fetch code.gitea.io/git` (+commit changes) to have the last commit of master and all file in sync.

File: vendor/vendor.json [hunk 1]
```diff
@@ -3,10 +3,10 @@
 	"ignore": "test appengine",
 	"package": [
 		{
-			"checksumSHA1": "wvujc6YaEoP3g2K+USjIcZcrRgQ=",
+			"checksumSHA1": "unD+SUYlIbPLpeMqVasIzsWbZPU=",
 			"path": "code.gitea.io/git",
-			"revision": "ed175468f8debc8cdb2fa495fc157f1b9d046628",
-			"revisionTime": "2017-05-04T06:58:26Z"
+			"revision": "ac69f9b0e2cde0dd064d6fa5055afd1fb2ec4d84",
```
Comment (sapk) [inline vendor/vendor.json]: This is the commit id in PR go-gitea/git#55 in master it is : go-gitea/git@56f6a2c9bb3bcdc8c9db64615e750016171ca7c6

**Comment by sapk** [comment]
Doing `govendor fetch code.gitea.io/git` I doesn't get the same hash. (don't know why ...)
```
[sapk@sapk-ultra gitea]$ govendor fetch code.gitea.io/git
[sapk@sapk-ultra gitea]$ git diff
diff --git a/vendor/vendor.json b/vendor/vendor.json
index 60e47f03..2a910aef 100644
--- a/vendor/vendor.json
+++ b/vendor/vendor.json
@@ -3,7 +3,7 @@
        "ignore": "test appengine",
        "package": [
                {
-                       "checksumSHA1": "/QkpRq7qiUkZzBbLFMZ0FFZYIOQ=",
+                       "checksumSHA1": "iqK8QebzwEcNL2iIRpMAOYwb7Zg=",
                        "path": "code.gitea.io/git",
                        "revision": "c98a6f353a13546ae13e7391020cadd750181585",
                        "revisionTime": "2017-05-29T11:48:12Z"
```

**Comment by lunny** [comment]
LGTM

**Comment by sapk** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Pull request merged**